lms/djangoapps/discussion/django_comment_client/utils.py

```diff
@@ -1040,3 +1046,54 @@ def sanitize_body(body):
     # This will remove the Markdown style links with data: URLs, and turn them
     # into empty links.
     return regex.sub(r'\]\(data:[^)]+\)', ']()', body)
+
+
+def convert_html_to_markdown(text):
+    """
+    Convert HTML to Markdown
+    """
+    text = convert_img_to_markdown(text)
+    text = convert_a_to_markdown(text)
+    text = convert_p_to_markdown(text)
+    return text
+
+
+def convert_a_to_markdown(text):
+    """
+    Convert <a> tags to Markdown format
+    """
+    def replace_link(match):
+        url = match.group(1)
+        if not url.startswith('http://') and not url.startswith('https://'):
+            url = 'https://' + url
+        title = match.group(3)
+        return f"[{title}]({url})"
+
+    # Use a regular expression to match each <a> tag and call replace_link on each match
+    # return re.sub(r'<a.*?href="(.*?)".*?title="(.*?)".*?</a>|<a.*?href="(.*?)".*?</a>', replace_link, text)
+    regex_exp = r'<a(?: [^>]+)? href="([^"]+)"(?: [^>]+)?(?: title="([^"]+)")?(?: [^>]+)?>(.*?)</a>'
+    return re.sub(regex_exp, replace_link, text)
+
+
+def convert_img_to_markdown(text):
+    """
+    Convert <img> tags to Markdown format
+    """
+    def replace_image(match):
+        url = match.group(1)
+        if not url.startswith('http://') and not url.startswith('https://'):
+            url = 'https://' + url
+        alt_text = match.group(2) or ''
+        title = match.group(3) or ''
+        return f"![{alt_text}]({url} \"{title}\")"
+
+    # Use a regular expression to match each <img> tag and call replace_image on each match
+    # pylint: disable=line-too-long
+    regex_exp = r'<img(?: [^>]+)? src="([^"]+)"(?: [^>]+)?(?: alt="([^"]+)")?(?: [^>]+)?(?: title="([^"]+)")?(?: [^>]+)?/>'
+
+    return re.sub(regex_exp, replace_image, text)
```

Comment:
Will re.sub work fine? I think if a post has 20-30 tags, it might take a huge amount of memory causing tab to crash. Can you test it by adding 40-50 img tags?

Reply:
Can you share any documentation for this, I am sure it can handle more than 30-50 images If there is a memory issue it would break earlier because of the size of the post and It might not happen because posts would not take memory more than a few Kbs. I don't think any post would take that much memory either.

And Images are not loaded on the backend if that is the concern only links to images are available in post data so basically it is in text form. In crux, it is just text manipulation.

Reply:
My bad. I meant it would break at the server and request will not complete. I am not sure the documentation will help in this case. I just remember a scenario in which 17 <p> tags caused a memory error on prod because of some computation being done using re.sub.

Reply:
Here is some test result with multiple links and multiple images . It is working fine.
https://user-images.githubusercontent.com/26253150/222664127-56b77090-086f-44f1-97db-37ffecfed58f.mov